azure/services/publicips/client.go

```diff
@@ -18,139 +18,104 @@ package publicips
 
 import (
 	"context"
-	"encoding/json"
 
-	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2021-08-01/network"
-	"github.com/Azure/go-autorest/autorest"
-	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v4"
 	"github.com/pkg/errors"
-	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-azure/azure"
+	"sigs.k8s.io/cluster-api-provider-azure/azure/services/asyncpoller"
 	"sigs.k8s.io/cluster-api-provider-azure/util/reconciler"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
 // AzureClient contains the Azure go-sdk Client.
 type AzureClient struct {
-	publicips network.PublicIPAddressesClient
+	publicips *armnetwork.PublicIPAddressesClient
+	auth      azure.Authorizer
```

Comment:
It looks like `auth` isn't used.

Reply:
Updated 7ba1619f6df44b7cc1ffb572c216a18297a7c4ed.